### DISCUSSION TITLE
ASIN Title, Price API Endpoint

### DISCUSSION THREAD

**Discussion by vipul-neon** [amzn/ads-advanced-tools-docs#96] [Q&A] [answered]
Hello,

Can you please mention which api endpoint must be called to get details about my brand's API? I tried calling product/metadata. however, it is always returning empty data

**Comment by vipul-neon** [comment]
@elizafarley could you please help here

**Comment by elizafarley** [comment]
Hi @vipul-neon, are you looking to see all SKU/ASINs associated with your advertising account? If yes, that can be accomplished using [POST /product/metadata](product-metadata/#/Product%20Selector/ProductMetadata) like this:

```
curl --location 'https://advertising-api.amazon.com/product/metadata' \
--header 'Amazon-Advertising-API-ClientId: amzn1.application-oa2-client.xxxxxxxxxxxx' \
--header 'Authorization: Bearer Atza|xxxxxxxxxxxx' \
--header 'Amazon-Advertising-API-Scope: xxxxxxxxxxx' \
--header 'Content-Type: text/plain' \
--data '{
   "pageSize": 10,
   "checkItemDetails": "true"
}'
```
Let me know if you are looking for other details about your brand, or that call isn't working.

**Reply by vipul-neon** [reply]
Hello,

Thank you for your response. I am making similar call but using python. I have pasted the code below. The result that I am getting is  -> b'{"ProductMetadataList":[]}'. Are there any additional access requirements for using this endpoint?
```
`# product_metadata

access_token = refresh_access_token(refresh_token)  #getting access token

Host = 'https://advertising-api-eu.amazon.com' #

params = { "pageSize":5, "checkItemDetails": "true"}

query = {"Amazon-Advertising-API-ClientId": lwa_client_id,
         "Amazon-Advertising-API-Scope":profile_id,
         "Authorization": f"Bearer {access_token}",
         'Content-Type': 'text/plain'
         }
response = requests.post(Host, headers = query, data = json.dumps(params))
sp_campaigns = response.content`
```

**Reply by vipul-neon** [reply]
It so happens that this endpoint is working for north American region and not for Europe region. 
@elizafarley please respond to this. Should my developer profile have access to brand API to fetch this information?

**Reply by elizafarley** [reply, marked as answer]
The profile you are calling from should only be authorized to use the associated regional endpoint. For example, if the profile ID is associated to the United States, you should only be able to call the NA endpoint. 

If you are getting an empty array in the response, it likely means there are no seller-owned ASINs associated to the profile.

**Reply by vipul-neon** [reply]
thanks for your help... i found the issues

**Comment by vipul-neon** [comment]
Dear @elizafarley,
I have one more question. could you be so kind to respond?

using the product/metadata endpoint, i am being able to fetch information for the clients who have Type = seller. 
However, the product medata list is empty for clients who have Type= vendor. 
Can you please help here?

**Reply by elizafarley** [reply]
Hi @vipul-neon, for vendors, product metadata fetches inventory based on the vendor code (an ID assigned to that profile ID). Advertisers can find this information in [Vendor Central](https://vendorcentral.amazon.com). If there are no ASINs associated to the vendor code, then no ASINs will return for that vendor.

**Reply by weixx57** [reply]
Hi @elizafarley ，we are also experiencing the same issue. When using the product/metadata API to retrieve product data, we are able to fetch data for some ASINs, but not for others. These ASINs are already listed and can be found in Vendor Central. Could you please advise on the possible reasons for this and how we should handle it? Thank you.

**Reply by elizafarley** [reply]
Hi @weixx57, that sounds like it could be a bug. I'd suggest [reaching out to support](https://amzn-clicks.atlassian.net/servicedesk/customer/portal/2/group/2) so the engineering team can investigate this issue.

**Reply by vipul-neon** [reply]
thanks @elizafarley for your responses. they are very helpful. have a good day!
